sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_usm.cpp

```diff
@@ -518,28 +528,81 @@ static ur_result_t USMAllocationMakeResident(
+    switch (BaseDesc->stype) {
+    case UR_STRUCTURE_TYPE_USM_POOL_LIMITS_DESC: {
+      const ur_usm_pool_limits_desc_t *Limits =
+          reinterpret_cast<const ur_usm_pool_limits_desc_t *>(BaseDesc);
+      for (auto &config : USMAllocatorConfigs.Configs) {
+        config.MaxPoolableSize = Limits->maxPoolableSize;
+        config.SlabMinSize = Limits->minDriverAllocSize;
+      }
+      break;
+    }
+    default: {
+      urPrint("urUSMPoolCreate: unexpected chained stype\n");
+      throw UsmAllocationException(UR_RESULT_ERROR_INVALID_ARGUMENT);
+    }
+    }
+    pNext = const_cast<void *>(BaseDesc->pNext);
+  }
+
+  HostMemPool = std::make_unique<USMAllocContext>(
+      std::unique_ptr<SystemMemory>(new USMHostMemoryAlloc(Context)),
+      this->USMAllocatorConfigs.Configs[usm_settings::MemType::Host]);
+
+  for (auto device : Context->Devices) {
+    DeviceMemPools[device] = std::make_unique<USMAllocContext>(
+        std::unique_ptr<SystemMemory>(
+            new USMDeviceMemoryAlloc(Context, device)),
+        this->USMAllocatorConfigs.Configs[usm_settings::MemType::Device]);
+
+    SharedMemPools[device] = std::make_unique<USMAllocContext>(
+        std::unique_ptr<SystemMemory>(
+            new USMSharedMemoryAlloc(Context, device)),
+        this->USMAllocatorConfigs.Configs[usm_settings::MemType::Shared]);
+    SharedMemReadOnlyPools[device] = std::make_unique<USMAllocContext>(
+        std::unique_ptr<SystemMemory>(
+            new USMSharedMemoryAlloc(Context, device)),
+        this->USMAllocatorConfigs
+            .Configs[usm_settings::MemType::SharedReadOnly]);
+  }
+}
+
 UR_APIEXPORT ur_result_t UR_APICALL urUSMPoolCreate(
     ur_context_handle_t Context, ///< [in] handle of the context object
     ur_usm_pool_desc_t
         *PoolDesc, ///< [in] pointer to USM pool descriptor. Can be chained with
                    ///< ::ur_usm_pool_limits_desc_t
     ur_usm_pool_handle_t *Pool ///< [out] pointer to USM memory pool
 ) {
-  std::ignore = Context;
-  std::ignore = PoolDesc;
-  std::ignore = Pool;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+
+  try {
+    *Pool = reinterpret_cast<ur_usm_pool_handle_t>(
+        new ur_usm_pool_handle_t_(Context, PoolDesc));
+  } catch (const UsmAllocationException &Ex) {
+    return Ex.getError();
+  }
+  return UR_RESULT_SUCCESS;
 }
 
 UR_APIEXPORT ur_result_t UR_APICALL urUSMPoolDestroy(
-    ur_context_handle_t Context, ///< [in] handle of the context object
-    ur_usm_pool_handle_t Pool    ///< [in] pointer to USM memory pool
+    [[maybe_unused]] ur_context_handle_t
+        Context,              ///< [in] handle of the context object
+    ur_usm_pool_handle_t Pool ///< [in] pointer to USM memory pool
 ) {
-  std::ignore = Context;
-  std::ignore = Pool;
-  urPrint("[UR][L0] %s function not implemented!\n", __FUNCTION__);
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+
```

Comment:
remove the maybe_unused and add it as std::ignore for uniformity with rest of code.

Reply:
Changed, but fwiw maybe_unused is the correct way to suppress an unused variable warning in c++17.